src/modules/launcher/Wox.Plugin/SharedCommands/DefaultBrowserInfo.cs

```diff
@@ -0,0 +1,235 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO.Abstractions;
+using System.Text;
+using Wox.Plugin.Logger;
+using Wox.Plugin.SharedCommands.Win32;
+
+namespace Wox.Plugin.SharedCommands
+{
+    /// <summary>
+    /// Contains information (e.g. path to executable, name...) about the default browser.
+    /// </summary>
+    public static class DefaultBrowserInfo
+    {
+        private static readonly object _updateLock = new object();
+        private static int _lastUpdateTickCount = -1;
+
+        public static readonly string MSEdgePath = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86) + @"\Microsoft\Edge\Application\msedge.exe";
+        public const string MSEdgeName = "Microsoft Edge";
+
+        private static readonly IPath FilePath = new FileSystem().Path;
+
+        /// <summary>Gets the path to default browser's executable.</summary>
+        public static string Path { get; private set; }
+
+        /// <summary>Gets <paramref name="Path" /> since the icon is embedded in the executable.</summary>
+        public static string IconPath { get => Path; }
+
+        public static string Name { get; private set; }
+
+        /// <summary>Gets a value indicating whether the browser supports querying a web search via command line argument (`? <term>`).</summary>
+        public static bool UseCmdLineArgumentForWebSearch { get => SearchEngineUrl is null; }
```

Comment:
```suggestion
        public bool SupportsWebSearchByCmdLineArgument { get => SearchEngineUrl is null; }
```

Sound more logically to me.

Reply:
theres a lock

Reply:
Aaaah. Now I see that the whole class is static. But let's rename it.

Reply:
yeah ill push in a bit